htdocs/core/class/html.form.class.php

```diff
@@ -3089,7 +3089,7 @@ public function select_produits_list($selected = 0, $htmlname = 'productid', $fi
 
 				// include search in supplier ref
 				if (getDolGlobalString('MAIN_SEARCH_PRODUCT_BY_FOURN_REF')) {
-					$sqlSupplierSearch .= !empty($sqlSupplierSearch) ? ' OR ':'';
+					$sqlSupplierSearch .= !empty($sqlSupplierSearch) ? ' AND ':'';
```

Comment:
Use of AND here looks strange.
If you search string "abc" you wan to get the line if ab is inside customer ref OR inside supplier ref.
Also having a AND will generate a incorrect condition 
a OR b OR c OR d AND ref_supplier
what is priority between OR and AND ?